### ISSUE TITLE
Transformation Reporting

### ISSUE THREAD

**Issue by pkbullock** [SharePoint/sp-dev-modernization#61]
#### Category

[ ] Bug
[X] Enhancement

#### Expected or Desired Behavior

When running the transformation process, recording the actions taken, success or failure, partial transforms. This can report out to the user/calling program as a report of the steps processed, any assumptions the application has taken, file asset locations, dependencies, issues and errors. 

This isn't referring to the scanner, just the transformer part of the application.

**Comment by jansenbe** [comment]
Fully agree. We should have logging and the logged data should be available from .Net as well as from file (depending on the log settings).

**Comment by pkbullock** [comment]
I think I may have some code to implement this quickly into the transformation engine. The code uses the observer pattern and allows you to register listeners so you can add Console, AppInsights, Custom ones - when you log events, you can notify the registered listeners accordingly.

Ill dig it out and see if it is a suitable for achieving this.

**Comment by pkbullock** [comment]
HI @jansenbe 

I have made quite a bit of headway with this and have a working POC. I have focused around one area of the transformation tool (PageTransformator) and used unit tests to get out, but the code additions are large again, think its worth creating a branch for this too.

I've realised in writing this, how much the transformation tool does to get a page converted to modern so kudos for all the work you have put into this.


Paul
